crates/resolver/src/lookup.rs

```diff
@@ -138,6 +139,32 @@ impl<'a> Iterator for LookupRecordIter<'a> {
     }
 }
 
+impl IntoIterator for Lookup {
+    type Item = RData;
+    type IntoIter = LookupIntoIter;
+
+    /// This is most likely not a free conversion, the RDatas will be cloned if data is
```

Comment:
Can there also be an implementation of `IntoIterator<Item = &'a Rdata>` for `&'a Lookup`? This could be implemented using the vec's [slice iterator](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.iter).

If I'm not missing anything, I think this should be possible?

Reply:
I think it's _also_ possible to have an iterator over a lookup where the lifetime of the `RData`s _doesn't_ depend on the lifetime of the `&Lookup` reference, by cloning the arc and manually indexing the vec. This might be more complex but it could have a nice usability benefit depending on the use-case?

I'm imagining something like this:

```
pub struct LookupIter<'a> {
    inner: Arc<Vec<Record>>,
    i: usize,
    _p: PhantomData<'a>,
}

impl<'a> Iterator for LookupIter<'a> {
    type Item = &'a RData;
    fn next(&mut self) -> Option<Self::Item> {
        let item = self.inner.get(self.i)?;
        self.i += 1;
        Some(item.unwrap_rdata())
    }
}
```

Reply:
Yes, thanks for your feedback. I was considering doing something like this. It would be better to not have the clone. Thanks for your feedback, I think I'll do this, and you even put in nearly the entire solution ;)

Reply:
I don't know why, but I don't always think of the PhantomData in resolving lifetime issues.

Reply:
Yeah, I feel like `PhantomData` is often a solution but it's not always obvious!

Reply:
So there is already an `iter` on `Lookup`. As far as I can tell, even with the PhantumData we need to constrain the `IntoIterator` impl to the `& Lookup`, which seems to possibly defeat the purpose of this change, as that could currently just have `IntoIterator` on that.

But this is slightly better in that we can delay the clone, and that might benefit `Copy` types like IpAddr.